enhancements/machine-config/pin-and-pre-load-images.md

```diff
@@ -336,11 +298,20 @@ garbage collection. To ensure disk space issues are reported and that garbage
 collection doesn't interfere we will introduced new mechanisms:
 
 1. Disk space will be checked before trying to pre-load images, and if there are
-issues they will be reported explicitly via the status of the `PinnedImageSet`
-status.
+issues they will be reported explicitly via the status of the
+`MachineConfigPool` status setting the pool as `Degraded`. A `MachineConfigPool`
+in `Degraded` state will block upgrades and this is by design. `PinnedImageSet`
+is a dependency of a cluster upgrade.
+
+1. There is a possibility that storage may become fully exhausted by unforeseen
+factors while images are being downloaded. Therefore, the configuration for
+`pinned_images` in CRI-O and the subsequent reloading of the CRI-O service are
```

Comment:
If I understand correctly, in current approach `prefetch_image_manager` will be the one who will write pinned_image list into `/etc/crio/crio.conf.d/50-pinned-images` once all the images has been pulled successfully on the node? If this is the case, maybe we should explicitly mention that in implementation detail.